hudi-spark-datasource/hudi-spark3.2.x/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -18,7 +18,9 @@
 package org.apache.spark.sql.avro
```

Comment:
It has to. The original logical (use `val converter: Any => Any = {`) has a bug that it will return the same value when we call this method twice continuously. And `HoodieCDCRDD` need these changes.

Reply:
> It has to. The original logical (use `val converter: Any => Any = {`) has a bug that it will return the same value when we call this method twice continuously. And `HoodieCDCRDD` need these changes.

so looks like an improvement we can land and fix separately? better to track it separately as CDC impl. do not need to know about this fix, right? the APIs remain the same. 

Another note: it's not very obvious to reviewers until you explained as above. so for the sake of faster review, please comment on it yourself and explain proactively.

Reply:
I'm not sure i understand what the issue is. Can you please create a separate PR with it? If there's a bug let's make sure we're adding necessary tests for it.